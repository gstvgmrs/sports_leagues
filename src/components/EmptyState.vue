<script setup lang="ts">
import Icon from "@/components/base/Icon.vue";

interface Props {
  icon?: 'document' | 'search' | 'close' | 'chevron-down' | 'error' | 'alert';
  title: string;
  description: string;
  iconClass?: string;
}

withDefaults(defineProps<Props>(), {
  icon: 'document',
  iconClass: 'mx-auto text-gray-300 mb-4'
});
</script>

<template>
  <div class="text-center py-12">
    <Icon :name="icon" size="lg" :class="iconClass" />
    <h3 class="text-lg font-medium text-gray-900 mb-2">{{ title }}</h3>
    <p class="text-gray-500">{{ description }}</p>
    <slot />
  </div>
</template>
