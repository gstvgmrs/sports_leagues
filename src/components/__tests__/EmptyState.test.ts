import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import EmptyState from '@/components/EmptyState.vue'

describe('EmptyState', () => {
  it('should match snapshot', () => {
    const wrapper = mount(EmptyState, {
      props: {
        icon: 'error',
        title: 'No leagues found',
        description: 'Try adjusting your search or filter criteria.',
        iconClass: 'text-red-500 custom-icon',
      },
      slots: {
        default: '<button class="mt-4 px-4 py-2 bg-blue-500 text-white rounded">Retry</button>',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
