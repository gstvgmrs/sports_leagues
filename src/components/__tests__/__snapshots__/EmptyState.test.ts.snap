// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyState > should match snapshot 1`] = `
"<div class="text-center py-12"><svg class="w-16 h-16 text-red-500 custom-icon" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4.5c-.77-.833-2.694-.833-3.464 0L3.34 16.5c-.77.833.192 2.5 1.732 2.5z"></path>
  </svg>
  <h3 class="text-lg font-medium text-gray-900 mb-2">No leagues found</h3>
  <p class="text-gray-500">Try adjusting your search or filter criteria.</p><button class="mt-4 px-4 py-2 bg-blue-500 text-white rounded">Retry</button>
</div>"
`;
